src/jit/unwind.cpp

```diff
@@ -199,30 +186,30 @@ void Compiler::unwindBegPrologCFI()
 
 void Compiler::unwindPushPopMaskCFI(regMaskTP regMask, bool isFloat)
 {
-    regMaskTP regBit = isFloat ? genRegMask(REG_FP_FIRST) : 1;
-
-    for (regNumber regNum = isFloat ? REG_FP_FIRST : REG_FIRST; regNum < REG_COUNT;
-         regNum           = REG_NEXT(regNum), regBit <<= 1)
+#if defined(_TARGET_ARM_)
+    for (regNumber regNum = isFloat ? REG_FP_LAST : REG_PC; regNum >= isFloat ? REG_FP_FIRST : REG_FIRST;
+         regNum           = REG_PREV(regNum))
+#else
+    for (regNumber regNum = isFloat ? REG_FP_FIRST : REG_FIRST; regNum < REG_COUNT; regNum = REG_NEXT(regNum))
+#endif
     {
-        if (regBit > regMask)
+        if (genRegMask(regNum) & regMask)
         {
-            break;
+            unwindPushPopCFI(regNum);
         }
-
-        if (regBit & regMask)
+// this is to prevent regNum underflow
```

Comment:
This is not pretty. It means that the for-loop condition is unreachable. Maybe ifdef out the whole loop for ARM? Something like:

```
#if defined(_TARGET_ARM_)
regNumber regNum = isFloat ? REG_FP_LAST : REG_PC; 
for (;;)
{
   if (genRegMask(regNum) & regMask)
   {
      unwindPushPopCFI(regNum);
   }

   if (regNum == isFloat ? REG_FP_FIRST : REG_FIRST)
   {
       break;
   }

   regNum = REG_PREV(regNum)
}
#else
```